src/modules/test/RecipeInvoker.cpp

```diff
@@ -25,42 +25,53 @@ void RecipeInvoker::TestBody()
-                                                                                                      << "JSON: " << payload;
+                    if (setting.second.type.compare("string") == 0)
+                    {
+                        EXPECT_NE(nullptr, json_object_get_string(jsonObject, setting.second.name.c_str())) << "Expecting '" << m_recipe.m_objectName << "' to contain string setting '" << setting.second.name << "'" << std::endl
+                                                                                                        << "JSON: " << payloadStr;
 
-                    std::string value = json_object_get_string(jsonObject, field.second.name.c_str());
-                    if (field.second.allowedValues->size() && std::find(field.second.allowedValues->begin(), field.second.allowedValues->end(), value) == field.second.allowedValues->end())
+                        std::string value = json_object_get_string(jsonObject, setting.second.name.c_str());
+                        if (setting.second.allowedValues->size() && std::find(setting.second.allowedValues->begin(), setting.second.allowedValues->end(), value) == setting.second.allowedValues->end())
+                        {
+                            FAIL() << "Field '" << setting.second.name << "' contains unsupported value '" << value << "'" << std::endl
+                                << "JSON: " << payloadStr;
+                        }
+                    }
+                    else if (setting.second.type.compare("integer") == 0)
                     {
-                        FAIL() << "Field '" << field.second.name << "' contains unsupported value '" << value << "'" << std::endl
-                               << "JSON: " << payload;
+                        JSON_Value *value = json_object_get_value(jsonObject, setting.second.name.c_str());
+                        EXPECT_EQ(JSONNumber, json_value_get_type(value)) << "Expecting '" << m_recipe.m_objectName << "' to contain integer setting '" << setting.second.name << "'" << std::endl
+                                                                        << "JSON: " << payloadStr;
+                    }
+                    else if (setting.second.type.compare("boolean") == 0)
+                    {
+                        JSON_Value *value = json_object_get_value(jsonObject, setting.second.name.c_str());
+                        EXPECT_EQ(JSONBoolean, json_value_get_type(value)) << "Expecting '" << m_recipe.m_objectName << "' to contain boolean setting '" << setting.second.name << "'" << std::endl
+                                                                        << "JSON: " << payloadStr;
+                    }
+                    else
+                    {
+                        FAIL() << "Unsupported type: " << setting.second.type << std::endl
+                            << "JSON: " << payloadStr;
```

Comment:
This fits on one single line. Don't break lines unnecessarily - here and in all other instances.